networkx/convert_matrix.py

```diff
@@ -1360,3 +1374,174 @@ def from_numpy_array(A, parallel_edges=False, create_using=None):
     return from_numpy_matrix(
         A, parallel_edges=parallel_edges, create_using=create_using
     )
+
+
+def generate_node_dataframe(
+    G: nx.Graph,
+    funcs: List[Callable],
+):
+    """
+    Return a pandas DataFrame representation of nodes and their metadata.
+
+    This function accepts a graph and a collection of user-defined functions
+    and returns a pandas DataFrame in which the nodes' are the DataFrame index,
+    the node metadata keys are column names,
+    and node metadata values are the values in the DataFrame.
+
+    ``funcs`` is an iterable of callables whose signature is
+
+        f(n, d) -> pd.Series
+
+    In other words, it accepts a (node, metadata dictionary) pair,
```

Comment:
```metadata dictionary``` is called ```nodeattr dict``` elsewhere... or more formally ```node attribute dict```

Reply:
Let me change that.